MonacoEditorComponent/Monaco/Editor/IModelDecorationOptions.cs

```diff
@@ -1,75 +1,110 @@
-﻿using Monaco.Helpers;
-using Newtonsoft.Json;
-using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Text;
-using System.Threading.Tasks;
-using Windows.Data.Json;
+﻿using Newtonsoft.Json;
 
 namespace Monaco.Editor
 {
     /// <summary>
-    /// Object Parser for https://microsoft.github.io/monaco-editor/api/interfaces/monaco.editor.imodeldecorationoptions.html
+    /// Options for a model decoration.
+    ///
+    /// Options associated with this decoration.
     /// </summary>
-    #pragma warning disable CS1591
     public sealed class IModelDecorationOptions
     {
-        [JsonProperty("isWholeLine")]
-        public bool IsWholeLine { get; set; }
+        /// <summary>
+        /// If set, the decoration will be rendered after the text with this CSS class name.
+        /// </summary>
+        [JsonProperty("afterContentClassName")]
+        public string AfterContentClassName { get; set; }
 
-        [JsonProperty("hoverMessage")]
-        public IMarkdownString[] HoverMessage { get; set; }
+        /// <summary>
+        /// If set, the decoration will be rendered before the text with this CSS class name.
+        /// </summary>
+        [JsonProperty("beforeContentClassName")]
+        public string BeforeContentClassName { get; set; }
 
-        [JsonProperty("glyphMarginHoverMessage")]
-        public IMarkdownString[] GlyphMarginHoverMessage { get; set; }
-
-        [JsonConverter(typeof(CssStyleConverter))]
+        /// <summary>
+        /// CSS class name describing the decoration.
+        /// </summary>
         [JsonProperty("className")]
-        public CssLineStyle ClassName { get; set; }
+        public string ClassName { get; set; }
```

Comment:
@hez2010 everything's looking really good. I've been playing with the PR and testing a bunch out even through a NuGet package test (even prepping a couple of other things I want to add for the release on top of it).

I think the last PR concern I have is the loss of strong-typing here on the C# side. What was your reasoning behind this part of the change?

Reply:
It's auto generated by tools. I can change it back to use `CssLineStyle`.

Reply:
I've added a commit to use `CssLineStyle`, `CssInlineStyle` and `CssGlyphStyle` for `ClassName`, `InlineClassName` and `GlyphMarginClassName`. However, what about `AfterContentClassName`, `BeforeContentClassName`, `LinesDecorationsClassName` and `MarginClassName`?

Reply:
Thanks @hez2010, much appreciated. My mantra for the project has just been to leverage the strong typing of C# and use C#/.NET/UWP types whenever possible to make it the most convenient for the C# developer leveraging components. 🙂

I think we're all set to merge. Anything else that I find in integration with my app, I'll fix forward before I push up a new NuGet package (have a couple of other features I want to try out).

It'd be nice to add a script later that can run the proper TypedocConverter command on the Monaco copy pulled by the other PowerShell Script. I may look at that quick next to grab some of the interfaces for a couple of the providers I want to add.